### PR TITLE
2326 delete tags

### PR DESCRIPTION
**Pull request by dgrove727**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/actions/DeleteContentTagAction.java [hunk 1]
```diff
@@ -103,4 +118,47 @@ protected void doAction(ActionEvent event) {
     protected void refreshDirectoryTree() {
     }
 
+    protected String getActionDisplayName() {
```
Comment (wschaeferB) [inline Core/src/org/sleuthkit/autopsy/actions/DeleteContentTagAction.java]: Getter appears to be unused so could be removed.

File: Core/src/org/sleuthkit/autopsy/actions/DeleteContentTagAction.java [hunk 2]
```diff
@@ -103,4 +118,47 @@ protected void doAction(ActionEvent event) {
...
+        return MENU_TEXT;
+    }
+
+    @NbBundle.Messages({"# {0} - fileID",
+            "DeleteContentTagAction.deleteTag.alert=Unable to untag file {0}."})
+    protected void deleteTag(TagName tagName, ContentTag contentTag, long fileId) {
```
Comment (wschaeferB) [inline Core/src/org/sleuthkit/autopsy/actions/DeleteContentTagAction.java]: Should actionPerformed or something else be calling this method or should it be removed

File: ImageGallery/src/org/sleuthkit/autopsy/imagegallery/actions/DeleteTagAction.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                        .log(Level.SEVERE, "Error retrieving tags for TagMenu", ex); //NON-NLS
+            }
+
+            if(getItems().isEmpty()) {
+                setDisable(true);
+            }
+        }
+    }
+
+    static private Window getIGWindow() {
```
Comment (wschaeferB) [inline ImageGallery/src/org/sleuthkit/autopsy/imagegallery/actions/DeleteTagAction.java]: Method is unused.

File: Core/src/org/sleuthkit/autopsy/actions/DeleteBlackboardArtifactTagAction.java [hunk 4]
```diff
@@ -20,26 +20,41 @@
...
+import org.sleuthkit.datamodel.TagName;
 import org.sleuthkit.datamodel.TskCoreException;
 
 /**
  * Instances of this Action allow users to delete tags applied to blackboard
  * artifacts.
  */
+@NbBundle.Messages({
+    "DeleteBlackboardArtifactTagAction.deleteTag=Delete Tag",
+    "DeleteBlackboardArtifactTagAction.unableToDelTag.msg=Unable to delete tag {0}.",
```
Comment (wschaeferB) [inline Core/src/org/sleuthkit/autopsy/actions/DeleteBlackboardArtifactTagAction.java]: Undocumented Format Parameter here and a few other places.

File: Core/src/org/sleuthkit/autopsy/datamodel/DataModelActionsFactory.java [hunk 5]
```diff
@@ -64,129 +71,227 @@
...
+        actionsList.add(null); // creates a menu separator
+        actionsList.add(ExtractAction.getInstance());
+        actionsList.add(null); // creates a menu separator
+        actionsList.add(AddContentTagAction.getInstance());
         if (isArtifactSource) {
-            actions.add(AddBlackboardArtifactTagAction.getInstance());
+            actionsList.add(AddBlackboardArtifactTagAction.getInstance());
+        }
+        
+        final Collection<AbstractFile> selectedFilesList =
```
Comment (wschaeferB) [inline Core/src/org/sleuthkit/autopsy/datamodel/DataModelActionsFactory.java]: Condensing these blocks of code into something like  
if(Utilities.actionsGlobalContext().lookupAll(AbstractFile.class).size() == 1){
 actionsList.add(DeleteFileContentTagAction.getInstance());
}
Would be possible, I don't know if it is more or less readable though so probably a matter of preference.

File: Core/src/org/sleuthkit/autopsy/actions/DeleteFileContentTagAction.java [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+            }
+        }.execute();
+    }
+
+    /**
+     * Instances of this class implement a context menu user interface for
+     * creating or selecting a tag name for a tag and specifying an optional tag
+     * comment.
+     */
+    // @@@ This user interface has some significant usability issues and needs
```
Comment (wschaeferB) [inline Core/src/org/sleuthkit/autopsy/actions/DeleteFileContentTagAction.java]: TODO style comment can be removed if story is created or issue was fixed

File: Core/src/org/sleuthkit/autopsy/actions/DeleteFileBlackboardArtifactTagAction.java [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+            }
+        }.execute();
+    }
+
+    /**
+     * Instances of this class implement a context menu user interface for
+     * creating or selecting a tag name for a tag and specifying an optional tag
+     * comment.
+     */
+    // @@@ This user interface has some significant usability issues and needs
```
Comment (wschaeferB) [inline Core/src/org/sleuthkit/autopsy/actions/DeleteFileBlackboardArtifactTagAction.java]: TODO style comment can be removed if story is created or issue was fixed

**Comment by dgrove727** [comment]
Wow, a lot of good points! I did a lot of cleanup, but it looks like there are still some things I missed here. I'll go ahead and get this updated.

**Comment by rcordovano** [comment]
Please target this PR to the release-4.4.0 branch, thanks.

**Pull request closed without merging**